### PR TITLE
fix #312 - no initial page load on Android

### LINKED ISSUE #312: initialPage does not work on Android with Hermes
> **Current Behavior**
> 
> `initialRouteName` don't work. Or scrolling to first route. Problem only on Android with Hermes (no problem on iOS with Hermes).
> 
> **Expected Behavior**
> 
> Tabs should open specified `initialRouteName` on init.
> 
> **How to reproduce**
> 
> https://github.com/Bardiamist/initialRouteRame
> 
> **Your Environment**
> 
> software	version
> iOS or Android	Android (Checked on Android 11 simulator)
> @react-navigation/native	5.9.3
> @react-navigation/material-top-tabs	5.3.14
> react-native-tab-view	3.0.0
> react-native	0.64.0
> expo	-
> node	15.12.0
> npm or yarn	yarn 1.22.10

Original post: https://github.com/react-navigation/react-navigation/issues/9435#issuecomment-802644853

### PR DESCRIPTION
**Pull request by Lukas-Heiligenbrunner**
# Summary

fix #312 with workaround solution from @KusStar (see https://github.com/callstack/react-native-pager-view/issues/312#issuecomment-807846238) 

## Checklist

- [X] I have tested this on a device and a simulator

### CODE REVIEW

File: src/PagerView.tsx [hunk 1]
```diff
@@ -59,10 +59,13 @@ export class PagerView extends React.Component<PagerViewProps> {
 
   componentDidMount() {
     // On iOS we do it directly on the native side
-    if (Platform.OS === 'android') {
-      if (this.props.initialPage != null) {
-        this.setPageWithoutAnimation(this.props.initialPage);
-      }
+    if (Platform.OS === 'android' && this.props.initialPage !== undefined) {
+      // que action to next repaint cycle
```
Comment (troZee) [inline src/PagerView.tsx:63]: please remove this comment

**Review by troZee — CHANGES_REQUESTED** [review]

**Review by troZee — APPROVED** [review]

File: src/PagerView.tsx (hunk 1 above)
Comment (Lukas-Heiligenbrunner) [reply src/PagerView.tsx]: why? :D

**Comment by danilobuerger** [comment]
@Lukas-Heiligenbrunner @troZee this doesn't really fix the issue. Now you can first see the first page before it skips to the `initialPage`.

**Comment by sajorahasan** [comment]
Yes, I'm facing the same issue, I can see 0 index first before jumping to initialPage index.

**Comment by troZee** [comment]
@danilobuerger @sajorahasan could you try this PR ? 

https://github.com/callstack/react-native-pager-view/pull/368

**Comment by rashidul0405** [comment]
> @Lukas-Heiligenbrunner @troZee this doesn't really fix the issue. Now you can first see the first page before it skips to the `initialPage`.

Same

**Comment by rashidul0405** [comment]
@troZee I just tested #368 and looks like the issue is fixed for android.

Step: `yarn remove react-native-pager-view && yarn add callstack/react-native-pager-view#368/head && ENVFILE=.env.development npx react-native run-android`

**Pull request merged**